src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -206,6 +217,22 @@ CommandResult executeIntroCommand() throws CommandException, ParseException {
         return commandResult;
     }
 
+    /**
+     * Executes the command to show the introduction. As the introduction command should not be
+     * accessible by the user, a commandText should not exist for it and the entire execution
+     * should be handled in the back-end. Note that as this method cannot throw a CommandException
+     * or a ParseException, handling can be ignored. The lack of an access modifier is intentional
+     * - method is supposed to be package private.
+     */
+
+    CommandResult executeShowReminderCommand() throws CommandException, ParseException {
+        CommandResult commandResult = logic.execute("showr");
+        logger.info("Result: " + commandResult.getFeedbackToUser());
+        resultPanel.setFeedbackToUser(commandResult.getFeedbackToUser());
+        handleReminders();
+        return commandResult;
+    }
+
```

Comment:
If it can't throw, the method should not have a throws keyword. Any other method that uses this are forced to implement catching the exceptions or throwing them again. Which is not the same as "this method cannot throw a CommandException or a ParseException, handling can be ignored".

In the case where command word for show reminder is changed but not changed here. This method **_will_** throw a exception. 

Do not follow the method above. Import enum CommandWord to avoid such incident.

Reply:
you realise that any command that uses `execute` has to either catch and deal with a `CommandException` and `ParseException` or it has to `throws` it as well?

Reply:
Also, it would be really helpful to say why something isn't a good implementation instead of brushing it aside as one.

Reply:
this information is outright incorrect. the fact that you've said something like this shows that you do not understand how the method works.

> If it can't throw, the method should not have a throws keyword.

correct.

> Which is not the same as "this method cannot throw a CommandException or a ParseException, handling can be ignored".

incorrect. because line 229 uses `Logic.execute()`, your method has to either implement a way to catch these exceptions by embedding it in a try-catch block, or simply put it in the signature. the problem with the former is that your catch block then cannot be empty, as if the exception is thrown by `Logic.execute()`, neither is a `CommandResult` returned nor an exception thrown. the reason you can put it in the signature is because, as the back-end programmer, you *know* that it will never throw this exception.

> In the case where command word for show reminder is changed in the enum but not changed here, this method will throw a exception.

yes, but `CommandException` and `ParseException` are both for users. as the programmer, you are expected to make consistent changes across the codebase when one reference changes. to use either of those exceptions as a way to ascertain that is wrong.

> Import enum CommandWord to avoid such incident. Implement catching and processing in case of an error, perhaps displaying an error message saying something like "reminders failed to show" in the resultPanel. 

this is literally what line 231 does.

> Do not follow executeIntroCommand() method above, its not a good implementation.

if you want to say this, at least have the courtesy to @ me.

Reply:
This is continued in person.